nix/sailcodegen.nix

```diff
@@ -0,0 +1,121 @@
+{ stdenv
+, ocamlPackages
+, fetchFromGitHub
+, lib
+, fetchMillDeps
+, mill
+, z3
+, gmp
+, zlib
+}:
+
+let
+  sailCodeGenSrc = with lib.fileset; toSource {
+    fileset = unions [
+      ../build.mill
+      ../common.mill
+      ../sailcodegen
+      ../rvdecoderdb
+    ];
+    root = ../.;
+  };
+  sailCodeGenDeps = fetchMillDeps {
+    name = "sailCodeGen";
+    src = sailCodeGenSrc;
+    millDepsHash = "sha256-j6ixFkxLsm8ihDLFkkJDePVuS2yVFmX3B1gOUVguCHQ=";
+  };
+  rvdecoderdbDeps = fetchMillDeps {
+    name = "rvdecoderdb";
+    src = sailCodeGenSrc;
+    millDepsHash = "sha256-j6ixFkxLsm8ihDLFkkJDePVuS2yVFmX3B1gOUVguCHQ=";
+  };
+in
+
+let
+  ocamlPackages' = ocamlPackages.overrideScope (finalScope: prevScope: {
+    sail = prevScope.sail.overrideAttrs rec {
+      version = "0.18";
+      src = fetchFromGitHub {
+        owner = "rems-project";
+        repo = "sail";
+        rev = version;
+        hash = "sha256-QvVK7KeAvJ/RfJXXYo6xEGEk5iOmVsZbvzW28MHRFic=";
+      };
+
+      buildInputs = [ finalScope.menhirLib ];
+    };
+  });
+  riscv-opcodes = {
```

Comment:
No need to copy paste `pname`, `version`, `date` attribute, just keep the src attribute here:

```nix
riscv-opcodes = fetchFromGitHub {
  # ...
}
```